test/data/repository/movie_repository_test.dart

```diff
@@ -0,0 +1,99 @@
+import 'package:the_movie_db_module2_part1/src/data/datasource/remote/i_api_service.dart';
+import 'package:the_movie_db_module2_part1/src/data/repository/movie_repository_impl.dart';
+import 'package:the_movie_db_module2_part1/src/domain/entity/movie.dart';
+import 'package:the_movie_db_module2_part1/src/core/util/movie_mocks.dart';
+
+class MockApiMovieService extends Mock implements IApiService {}
+
+void main() {
+  late MovieRepositoryImpl movieRepo;
+  late MockApiMovieService mockMovieService;
+
+  setUp(() {
+    mockMovieService = MockApiMovieService();
+    movieRepo = MovieRepositoryImpl(
+      movieService: mockMovieService,
+    );
+  });
+
+  group(
+    'Repository getData() method',
+    () {
+      test(
+        'Movie service is called properly',
+        () async {
+          when(() => mockMovieService.fetchDataFromApi()).thenAnswer(
+            (_) async => DataSuccess(
+              MovieMocks.mockedMovieModelList,
+            ),
+          );
+          await movieRepo.getData();
+          verify(() => mockMovieService.fetchDataFromApi()).called(1);
+        },
+      );
+
+      test(
+        'getData() successfully gets movies,'
+        'gets DataSuccess from service',
+        () async {
+          when(() => mockMovieService.fetchDataFromApi()).thenAnswer(
+            (_) async => DataSuccess(
+              MovieMocks.mockedMovieModelList,
+            ),
+          );
+          DataState<List<Movie>> movies = await movieRepo.getData();
+          expect(movies, isInstanceOf<DataSuccess>());
+          expect(movies.data?.first, isA<Movie>());
+        },
+      );
+
+      test(
+        'getData() receives DataSuccess with an empty model,'
+        'should return DataError and throw Exception ',
+        () async {
+          when(() => mockMovieService.fetchDataFromApi()).thenAnswer(
+            (_) async => const DataSuccess(
+              MovieMocks.emptyMovieModelList,
+            ),
+          );
+          try {
+            DataState movies = await movieRepo.getData();
+            expect(movies, isInstanceOf<DataError>());
+          } on Exception catch (e) {
+            expect(
+              e.toString(),
+              contains(ApiConstants.errorMessage),
+            );
+          }
+        },
+      );
+
+      test(
+        'getData() receives a DataError,'
+        'should throw exception',
+        () async {
+          when(() => mockMovieService.fetchDataFromApi()).thenAnswer(
+            (_) async => DataError(
+              Exception(
+                ApiConstants.errorMessage,
+              ),
+            ),
+          );
+          try {
+            DataState movies = await movieRepo.getData();
+            expect(movies, isInstanceOf<DataError>());
+          } on Exception catch (e) {
+            expect(
+              e.toString(),
+              contains(ApiConstants.errorMessage),
+            );
+          }
```

Comment:
To test exceptions it's better to:
```dart
expect(() => operations.lookupOrderDetails(), throwsA(isA<MyCustErr>()));
```
But does the repo raises an exception or returns a `DataError`? I don't understand that based on your test.

Reply:
you are right, my repo is returning a DataError, not throwing an exception, also the same in the use cases, i'll change it!